types/txMsg.ts

```diff
@@ -81,3 +84,13 @@ export interface TxMsgCreateVestingAccount {
     readonly delayed: boolean;
   };
 }
+
+type OmmitedMsgTransfer = Omit<MsgTransfer, "timeoutHeight" | "memo">;
+type MsgTransferRequiredToken = OmmitedMsgTransfer & Required<Pick<OmmitedMsgTransfer, "token">>;
+interface MsgTransferOptionalMemo extends MsgTransferRequiredToken {
+  readonly memo?: string;
```

Comment:
Why not keep this required and just set it empty when the user does not put in text?

Reply:
This is a problem coming from Go and protobuf. There the something is called "optional" wen it is allowed to be empty. Since at transaction level `unset` and `""` are the same thing we don't need to differentiate it here.